modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -262,7 +262,8 @@ export class {{classname}} {
 {{#hasFormParams}}
         const canConsumeForm = this.canConsumeForm(consumes);
 
-        let formParams: { append(param: string, value: any): void; };
+        let formParams: FormParams;
```

Comment:
formParams could be an `HttpParams` and need to not use the return as the same way

Reply:
@emmanuelgautier `FormParams` type is defined [here](https://github.com/swagger-api/swagger-codegen/pull/9012/files#diff-5bdae26ba753c4986ceb5af35bada047R364) as a union type that includes HttpParams, so I think that would work. What am I missing?
What do you mean by "and need not use the return as the same way"?

Reply:
`FormData` and `HttpParams` append function does not have the same signature :
- [HttpParams](https://angular.io/api/common/http/HttpParams#append)
- [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append)

One return FormParams and the other not. When you use `formParams.append`, you have to check what is the exact variable type. Here, you do not make this check and could be a potential error when formParams is a HttpParams and you use HttpClient

It is almost the same thing that I done in this [lines](https://github.com/swagger-api/swagger-codegen/pull/9083/files#diff-5bdae26ba753c4986ceb5af35bada047R266).

Reply:
@emmanuelgautier: They differ on the return type, that is the reason why in my method, I implicitly check the type (by checking the `useReturnValue `) - 
If it's `HttpParams` I return the return value of `append()`, if it isn't, I append, and return the same instance (see the comma `,` [there](https://github.com/swagger-api/swagger-codegen/pull/9012/files#diff-5bdae26ba753c4986ceb5af35bada047R361)).

Reply:
Here, if you use HttpClient, you do not use `useReturnValue` but formParams could be a `HttpParams` and return value.

Reply:
You are right. My bad 👍